src/Parser/Parser.hs

```diff
@@ -358,32 +361,36 @@ btype =
   --reserved "Input" *> pure Input
   <|>
   reserved "Board" *> pure Board
-  -- removes 'AnySymbol' as a directly usable type
-  -- serves as a parent of all types internally, but this prevents
-  -- it from being referenced externally (as requested by the CSforAll group) (@montymxb)
-  -- <|>
-  -- reserved "AnySymbol" *> pure AnySymbol
 
 -- | Parse an enum of a set of names
 enum :: Parser (S.Set Name)
 enum = reservedOp "{" *>
          (S.fromList <$> (commaSep1 (notAlreadyInUse capIdentifier))) <* reservedOp "}"
 
+-- | Bad extension error message
+badExtension :: String
+badExtension = "the right side of & must either be an {Enumeration} or the name of one"
+
+-- | Dereference a named enumeration type and union its symbols
+chase :: Name -> Parser (S.Set Name)
+chase n = do
+  t <- (lookup <$> (pure n) <*> (tdef <$> getState))
+  case t of
+    Nothing -> fail $ "Type " ++ n ++ " not declared!"
```

Comment:
Note that due to the syntax of types, the only way to store extended types made up of multiple enumerated types is as a single enumerated type. For example: `{A, B} & {C, D}` can only be stored as `{A, B, C, D}`, hence the union and chase stuff below.